testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java

```diff
@@ -0,0 +1,137 @@
+import static io.quarkus.ts.startstop.utils.Commands.getBuildCommand;
+import static io.quarkus.ts.startstop.utils.Commands.parsePort;
+import static io.quarkus.ts.startstop.utils.Commands.processStopper;
+import static io.quarkus.ts.startstop.utils.Commands.runCommand;
+import static io.quarkus.ts.startstop.utils.Commands.unzip;
+import static io.quarkus.ts.startstop.utils.Commands.waitForTcpClosed;
+import static io.quarkus.ts.startstop.utils.Logs.archiveLog;
+import static io.quarkus.ts.startstop.utils.Logs.checkLog;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author Michal Karm Babacek <karm@redhat.com>
+ */
+@Tag("codequarkus")
+public class CodeQuarkusTest {
+
+    private static final Logger LOGGER = Logger.getLogger(CodeQuarkusTest.class.getName());
+
+    public static final String GEN_BASE_DIR = getArtifactGeneBaseDir();
+
+    public static final List<List<CodeQuarkusExtensions>> supportedEx = CodeQuarkusExtensions.partition(4, CodeQuarkusExtensions.Flag.SUPPORTED);
+    public static final List<List<CodeQuarkusExtensions>> notSupportedEx = CodeQuarkusExtensions.partition(1, CodeQuarkusExtensions.Flag.NOT_SUPPORTED);
+    public static final List<List<CodeQuarkusExtensions>> mixedEx = CodeQuarkusExtensions.partition(1, CodeQuarkusExtensions.Flag.MIXED);
+
+    public void testRuntime(TestInfo testInfo, List<CodeQuarkusExtensions> extensions) throws Exception {
+        Process pA = null;
+        File unzipLog = null;
+        File runLogA = null;
+        String cn = testInfo.getTestClass().get().getCanonicalName();
+        String mn = testInfo.getTestMethod().get().getName();
+        LOGGER.info(mn + ": Testing Code Quarkus generator with these " + extensions.size() + " extensions: " + extensions.toString());
+        File appDir = new File(GEN_BASE_DIR + File.separator + "code-with-quarkus");
+        String logsDir = GEN_BASE_DIR + File.separator + "code-with-quarkus-logs";
+        List<String> devCmd = getBuildCommand(MvnCmds.MVNW_DEV.mvnCmds[0]);
+        URLContent skeletonApp = Apps.GENERATED_SKELETON.urlContent;
+        String zipFile = GEN_BASE_DIR + File.separator + "code-with-quarkus.zip";
+
+        try {
+            cleanDir(appDir.getAbsolutePath(), logsDir);
+            Files.createDirectories(Paths.get(logsDir));
+            LOGGER.info("Downloading...");
+            download(extensions, zipFile);
+            LOGGER.info("Unzipping...");
+            unzipLog = unzip(zipFile, GEN_BASE_DIR);
+            LOGGER.info("Running...");
+            runLogA = new File(logsDir + File.separator + "dev-run.log");
+            pA = runCommand(devCmd, appDir, runLogA);
+            // It takes time to download the Internet
+            long timeoutS = 30 * 60;
+            LOGGER.info("Waiting for the web content...");
+            WebpageTester.testWeb(skeletonApp.urlContent[0][0], timeoutS, skeletonApp.urlContent[0][1], false);
+            LOGGER.info("Terminating and scanning logs...");
```

Comment:
While I was adding my comments I also waited for 30 minutes timeout to see what happens.

There is no warn or error message about timeout being reached, just moving on to the next test.
Add message when the timeout is reached

```
Apr 11, 2020 4:40:38 PM io.quarkus.ts.startstop.CodeQuarkusTest testRuntime
INFO: Unzipping...
Apr 11, 2020 4:40:38 PM io.quarkus.ts.startstop.CodeQuarkusTest testRuntime
INFO: Running...
Apr 11, 2020 4:40:38 PM io.quarkus.ts.startstop.CodeQuarkusTest testRuntime
INFO: Waiting for the web content...
Apr 11, 2020 5:10:39 PM io.quarkus.ts.startstop.CodeQuarkusTest testRuntime
INFO: notSupportedExtensions: Testing Code Quarkus generator with these 10 extensions: [QUARKUS_RESTEASY_MUTINY, QUARKUS_RESTEASY_QUTE, QUARKUS_JDBC_H2, QUARKUS_JDBC_DERBY, QUARKUS_AMAZON_DYNAMODB, QUARKUS_CACHE, QUARKUS_FLYWAY, QUARKUS_HIBERNATE_SEARCH_ELASTICSEARCH, QUARKUS_INFINISPAN_CLIENT, QUARKUS_LIQUIBASE]
```

Reply:
```
INFO: Timeout 600s was reached. Empty webpage does not contain string: `Congratulations'

```